src/main/java/frc/robot/commands/conveyor/ConveyorAutomation.java

```diff
@@ -0,0 +1,93 @@
+package frc.robot.commands.conveyor;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.devices.ColorSensor;
+import frc.robot.devices.LidarV3;
+
+/**
+ * Command to track the possible states of positions within the conveyor.
+ */
+public class ConveyorAutomation extends CommandBase {
+
+    /**
+     * Enum tracking what could be in the front or back of the conveyor.
+     */
+    public enum ConveyorState {
+        NONE,
+        BLUE,
+        RED
+    }
+    
+    private final ColorSensor colorSensor;
+    private final LidarV3 lidar;
+    private ConveyorState frontState;
+    private ConveyorState backState;
+    private String previousColorSensorMeasurement;
+    private String colorSensorMeasurement;
+    private double lidarDistance;
+
+    private static final double
+        MIN_LIDAR_DISTANCE = 0,
+        MAX_LIDAR_DISTANCE = 0;
+
+    /**
+     * Command to track the possible states of positions within the conveyor.
+     *
+     * @param colorSensor the color sensor
+     * @param lidar the lidar
+     */
+    public ConveyorAutomation(ColorSensor colorSensor, LidarV3 lidar) {
+        this.colorSensor = colorSensor;
+        this.lidar = lidar;
+
+        frontState = ConveyorState.NONE;
```

Comment:
move this setup stuff from the constructor to initialize as the constructor will not be called if the robot is disabled and then re-enabled even though the balls may have changed